device/bgp/bgpsrtesegment.yaml

```diff
@@ -3,47 +3,490 @@ components:
+      description: >-
+        SRv6 Endpoint Behavior and SID Structure and its optional. 
+        The total of the Locator Block, Locator Node, Function, and Argument
+        lengths MUST be less than or equal to 128.
+      type: object
+      properties:
+        lb_length:
+          description: >-
+            SRv6 SID Locator Block length in bits.
+          type: integer
+          minimum: 0
+          maximum: 128
           default: 0
-        mpls_ttl:
+        ln_length:
           description: >-
-            TBD
+            SRv6 SID Locator Node length in bits.
           type: integer
+          minimum: 0
+          maximum: 128
           default: 0
-        v_flag:
+        func_length:
           description: >-
-            TBD
-          type: boolean
-          default: false
-        ipv6_sid:
+            SRv6 SID Function length in bits.
+          type: integer
+          minimum: 0
+          maximum: 128
+          default: 0
+        arg_length:
+          description: >-
+            SRv6 SID Arguments length in bits.
+          type: integer
+          minimum: 0
+          maximum: 128
+          default: 0
+    
+    BgpSrte.SegmentATypeSubTlv:
+      x-include:
+      - '.#/components/schemas/BgpSrte.SrMplsSid'
+      description: >-
+        Type  A: SID only, in the form of MPLS Label.
+      type: object
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
+            https://datatracker.ietf.org/doc/html/draft-ietf-idr-segment-routing-te-policy-13#section-2.4.4.2.12
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        
+    BgpSrte.SegmentBTypeSubTlv:
+      description: >-
+        Type  B: SID only, in the form of IPv6 address.
+      type: object
+      required: [srv6_sid]
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
+            https://datatracker.ietf.org/doc/html/draft-ietf-idr-segment-routing-te-policy-13#section-2.4.4.2.12
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        srv6_sid:
+          description: >-
+            SRv6 SID.
+          type: string
+          format: ipv6
+        srv6_sid_endpoint_behavior:
+          description: >-
+            Optional SRv6 Endpoint Behavior and SID Structure.
+          $ref: '#/components/schemas/BgpSrte.SRv6SIDEndpointBehaviorAndStructure'
+    
+    BgpSrte.SegmentCTypeSubTlv:
+      x-status: under-review
+      description: >-
+        Type C: IPv4 Node Address with optional SID.
+      type: object
+      required: [ipv4_node_address]
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
```

Comment:
One octet bitmap for flags including V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in